src/main/java/com/monitorjbl/plugins/config/UserResource.java

```diff
@@ -31,12 +30,12 @@ public Response get(@PathParam("projectKey") String projectKey, @PathParam("repo
     return Response.ok(ImmutableMap.of(
         "requiredReviews", config.getRequiredReviews() == null ? "" : config.getRequiredReviews(),
         "blockMergeIfPrNeedsWork", config.getBlockMergeIfPrNeedsWork() == null ? "" : config.getBlockMergeIfPrNeedsWork(),
-        "requiredReviewers", utils.dereferenceUsers(newArrayList(newHashSet(concat(
```

Comment:
This code was loading the dereferencing the groups, which implicitly returns "dereferenced" users by name, and then dereferencing those users _again_ together with any explicitly configured users. I've changed it to not re-dereference users from dereferenced groups. Instead, it only dereferences explicitly configured users.

(Note that this code could, and still can, return duplicates, if a user was in a configured group _and_ configured by name.)

Reply:
The HashSet was guarding against duplicates. Was there a reason you removed that part?

Reply:
Ah, of course. You were deduping on the _input_ side, before. The old code didn't have any deduping on the _output_, but since you eliminated them from the input it wouldn't matter.

I'll re-add the set, but around the user/group call instead of inside. That way group users still don't get dereferenced twice.

Reply:
Turns out I'd introduced a more subtle breakage than that, in that `dereferenceGroups` and `dereferenceUsers` return different types (Strings and `User`s, respectively). I've fixed that now too.

However, noticing _that_ led me to notice a different bug. Specifically, the `String`s returned by `dereferenceGroups` are user _slugs_, not user _names_. But in `CommitBlockerHook`, they're being compared to `UserProfile.getUsername()`. That means in any instance where users' names and slugs are not identical, configuring excluded groups will not actually work. So I fixed that too.

The _best_ fix for that is to replace references to SAL's `UserManager` with Bitbucket Server's `AuthenticationContext`, which returns an `ApplicationUser` directly (and allows access to its slug). Using `AuthenticationContext` also simplifies `ConfigServlet` and removes the need for `UserUtils.getApplicationUserByName`, so I removed that too.